### DISCUSSION TITLE
Why is the same event removed and added on the same object?

### DISCUSSION THREAD

**Discussion by scotch83** [xamarin/XamarinCommunityToolkit#1929] [General]
In this method there is a line where we remove an event which is re-added right after...why is that?
https://github.com/xamarin/XamarinCommunityToolkit/blob/299fac61a991d5ed0aac925b07e404fcb22c6f3e/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/TabView/TabView.shared.cs#L521
https://github.com/xamarin/XamarinCommunityToolkit/blob/299fac61a991d5ed0aac925b07e404fcb22c6f3e/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/TabView/TabView.shared.cs#L522

**Comment by bijington** [comment]
We unsubscribe before subscribing again to make sure we don't end up with multiple subscriptions to the same event handler.
